docker/provider.go

```diff
@@ -260,13 +266,18 @@ func parseDockerConfig(r io.Reader) (map[string]dockerConfig, error) {
 	return confs, nil
 }
 
-// authConfigs converts a dockerConfigs map to a AuthConfigs object.
-func authConfigs(confs map[string]dockerConfig) (*AuthConfigs, error) {
+// convertDockerConfigToAuthConfigs converts a dockerConfigs map to a AuthConfigs object.
+func convertDockerConfigToAuthConfigs(confs map[string]dockerConfig) (*AuthConfigs, error) {
 	c := &AuthConfigs{
 		Configs: make(map[string]types.AuthConfig),
 	}
-	for reg, conf := range confs {
+	for registryAddress, conf := range confs {
 		if conf.Auth == "" {
+			authFromKeyChain, err := getCredentialsFromOSKeychain(registryAddress)
+			if err != nil {
+				return nil, err
+			}
+			c.Configs[registryAddress] = authFromKeyChain
```

Comment:
There won't be a problem because later we iterate over the `Configs` and check if we find credentials for a given `registry`. If there are no credentials found we use the defaults of `types.AuthConfig` which are empty username and password. So the auth won't work